### PR TITLE
feat(app): update deck config in device details for run status

### PR DESCRIPTION
**Pull request by koji**
# Overview
Add a new banner and make DeckConfigurator disable when run is running

Screenshot 2023-10-27 at 4 51 37 PM

close RAUT-824

# Test Plan
- run a protocol and go to device details page
check two things banner is displayed and deck configurator is not clickable

# Changelog

# Review requests

# Risk assessment
low

### CODE REVIEW

File: app/src/organisms/DeviceDetailsDeckConfiguration/__tests__/DeviceDetailsDeckConfiguration.test.tsx [hunk 1]
```diff
@@ -67,4 +79,13 @@ describe('DeviceDetailsDeckConfiguration', () => {
...
     getByText('mock DeckFixtureSetupInstructionsModal')
   })
+
+  it('should render banner and make deck configurator disabled when running', () => {
+    RUN_STATUSES.isRunRunning = true
+    mockUseRunStatuses.mockReturnValue(RUN_STATUSES)
+    const [{ getByText, queryAllByRole }] = render(props)
+    getByText('Deck configuration is not available when run is in progress')
+    // Note (kk:10/27/2023) detects Setup Instructions buttons
+    expect(queryAllByRole('button').length).toBe(1)
```
Comment (koji) [inline app/src/organisms/DeviceDetailsDeckConfiguration/__tests__/DeviceDetailsDeckConfiguration.test.tsx:89]: At this moment, I don't have any better ideas to check DeckConfigurator is disabled.
If there is any better or proper way, please let me know.

File: app/src/organisms/DeviceDetailsDeckConfiguration/index.tsx [hunk 2]
```diff
@@ -117,55 +120,66 @@ export function DeviceDetailsDeckConfiguration({
...
-
         <Flex
-          gridGap={SPACING.spacing40}
+          gridGap={SPACING.spacing16}
           paddingX={SPACING.spacing16}
-          paddingY={SPACING.spacing32}
+          paddingBottom={SPACING.spacing32}
+          paddingTop={isRunRunning ? undefined : SPACING.spacing32}
           width="100%"
+          id="hello"
```
Comment (jerader) [inline app/src/organisms/DeviceDetailsDeckConfiguration/index.tsx]: maybe you forgot to remove this id

File: app/src/organisms/DeviceDetailsDeckConfiguration/__tests__/DeviceDetailsDeckConfiguration.test.tsx (hunk 1 above)
Comment (jerader) [reply app/src/organisms/DeviceDetailsDeckConfiguration/__tests__/DeviceDetailsDeckConfiguration.test.tsx:89]: this seems okay to me, but maybe @brenthagen has some more ideas! The other way to test if certain buttons are disabled is if you don't mock deck configurator and check one of the buttons on a fixture.

Comment (brenthagen) [reply app/src/organisms/DeviceDetailsDeckConfiguration/__tests__/DeviceDetailsDeckConfiguration.test.tsx:89]: i think mocking the DeckConfigurator with `jest-when` and a specific read only mock return value for `partialComponentPropsMatcher({ readOnly: true })` should work

**Review by brenthagen — APPROVED** [review]
looks good to me !

**Comment by koji** [comment]
Tried to jest-when but it didn't work as expected. I will update this test later.

**Pull request merged**